storage/client.go

```diff
@@ -295,6 +296,19 @@ func (c Client) exec(verb, url string, headers map[string]string, body io.Reader
 	}
 
 	req, err := http.NewRequest(verb, url, body)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Can you add a bit detail before the err? It might look awkward upper in the stack and it light be hard to tell where this is coming from.